portfolio/src/main/webapp/script.js

```diff
@@ -16,138 +16,162 @@
-    event.currentTarget.className += " active";
-
-    if (tabName == "Interests"){
-        drawTimeline();
-    }
+  hideMoreInfo();
+  
+  var buttonContainer = document.getElementsByClassName("tabs")[0];
+  buttonContainer.style.borderBottomLeftRadius = "0";
+  buttonContainer.style.borderBottomRightRadius = "0";
+
+  var tabInfo = document.getElementsByClassName("tabinfo");
+  for (var i = 0; i < tabInfo.length; i++) {
+    tabInfo[i].style.display = "none";
+  }
+
+  var tabButtons = document.getElementsByClassName("tabbutton");
+  for (var i = 0; i < tabButtons.length; i++) {
+    tabButtons[i].className = tabButtons[i].className.replace(" active", "");
+  }
+
+  document.getElementById(tabName).style.display = "block";
+  event.currentTarget.className += " active";
+
+  if (tabName == "Interests"){
+    drawTimeline();
+  }
 }
 
 /*
  * Displays the timeline animation on the Interests tab.
  */
 function drawTimeline() {
-    hideImages();
-    hideBranches();
-    var timeline = document.getElementById("timeline");
-    var movingDot = document.getElementById("moving-dot")
-    var rightEnd = 95;
-    var dotLeftPos = 97 - rightEnd;
-    var branchPosToExtend = 0;
-
-    var branchTriggers = {
-        6: "astronomy",
-        22: "geography",
-        38: "tornadoes",
-        54: "architecture",
-        70: "photography",
-        86: "compsci"
-    }
-
-    var id = setInterval(extendTimeline, 20);
-    function extendTimeline() {
-        if (rightEnd <= 5) {
-            clearInterval(id);
-        } else {
-            rightEnd -= 0.5;
-            dotLeftPos = 97 - rightEnd;
-            timeline.style.right = rightEnd + "%";
-            movingDot.style.left = dotLeftPos + "%";
-
-            if (dotLeftPos in branchTriggers) {
-                displayImage(branchTriggers[dotLeftPos]);
-                displayTimelineBranch(branchTriggers[dotLeftPos], branchPosToExtend);
-                branchPosToExtend = 1 - branchPosToExtend;
-            }
-        }
+  //branches and the images to their starting positions
+  hideImages();
+  hideBranches();
+
+  var timeline = document.getElementById("timeline");
+  var movingEndpoint = document.getElementById("moving-endpoint")
+
+  //moves the right end of the timeline, resulting in the left-to-right animation
+  var timelineRightPosition = 95;
+
+  //moves the "Present" endpoint to the right as the timeline expands
+  var endpointLeftPosition = 97 - timelineRightPosition;
+
+  //controls the direction of each branch's animation (true expands up, false expands down)
+  var branchUp = true;
+
+  //triggers each branch's animation at the appropriate point in the timeline's expansion
+  var branchTriggers = {
+    6: "astronomy",
+    22: "geography",
+    38: "tornadoes",
+    54: "architecture",
+    70: "photography",
+    86: "compsci"
+  }
+
+  var timelineAnimation = setInterval(extendTimeline, 20);
+  function extendTimeline() {
+    //stops the timeline's animation
```

Comment:
I am not able to find an external documentation right now, but comments should follow the following rules:
1. They must begin within a capital letter.
2. They must end in a period.
3. There should be a space between // and the first character of the comment.